### PR TITLE
Configure Github Actions

### PR DESCRIPTION
**Pull request by maxlambrecht**
This PR adds the config to run the Gradle build with Github Actions in multiple environments (Ubuntu and MacOs, with Java 8, 11 and 16)
Gradle and Lombok dependency were updated to support building with Java 16.
It also removes the travis configuration.

Signed-off-by: Max Lambrecht <maxlambrecht@gmail.com>

### CODE REVIEW

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - name: Build with Gradle and generate the artifacts
+        run: ./gradlew assemble test
+      - uses: actions/upload-artifact@v2
+        with:
+          name: Package
+          path: build/libs
```
Comment (rturner3) [inline .github/workflows/gradle.yml:35]: Do we need to remove these Gradle files from the cache as a final step in the build?
```
~/.gradle/caches/modules-2/modules-2.lock
~/.gradle/caches/modules-2/gc.properties
```

I noticed a [comment about it](https://docs.github.com/en/actions/guides/building-and-testing-java-with-gradle#caching-dependencies) in the GitHub documentation.

The first file was also removed by the previous Travis build.

File: .github/workflows/gradle.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Build with Gradle
+        run: ./gradlew build
```
Comment (rturner3) [inline .github/workflows/gradle.yml:48]: It would be nice to also run tests for each JDK + OS combination.

**Review by rturner3 — COMMENTED** [review]
Do we have any way to collect code coverage statistics in the GitHub actions CI builds? I saw that it used to be set up for Travis.

File: .github/workflows/gradle.yml (hunk 1 above)
Comment (maxlambrecht) [reply .github/workflows/gradle.yml:35]: > Do we need to remove these Gradle files from the cache as a final step in the build?
> 
> ```
> ~/.gradle/caches/modules-2/modules-2.lock
> ~/.gradle/caches/modules-2/gc.properties
> ```
> 
> I noticed a [comment about it](https://docs.github.com/en/actions/guides/building-and-testing-java-with-gradle#caching-dependencies) in the GitHub documentation.
> 
> The first file was also removed by the previous Travis build.

hmm yeah, we would probably need to delete those. I'll add that step.

**Comment by maxlambrecht** [comment]
> Do we have any way to collect code coverage statistics in the GitHub actions CI builds? I saw that it used to be set up for Travis.

That's a very good point! I didn't look into it, I'll see if I can add that as part of this PR.

File: .github/workflows/gradle.yml (hunk 2 above)
Comment (maxlambrecht) [reply .github/workflows/gradle.yml:48]: Which JDK versions would you consider relevant?

Comment (rturner3) [reply .github/workflows/gradle.yml:48]: Whichever ones we claim support for - today I believe that's 8, 11, and 16?

Comment (rturner3) [reply .github/workflows/gradle.yml:48]: Whichever ones are actively supported by Oracle tends to be a reasonable set of versions to maintain compatibility for. Today that would be 8 (LTS), 11 (LTS), and 16 - the same ones you have here.

I now recall that the Gradle `build` target also runs tests, so what you have here looks good!

**Comment by maxlambrecht** [comment]
> > Do we have any way to collect code coverage statistics in the GitHub actions CI builds? I saw that it used to be set up for Travis.
> 
> That's a very good point! I didn't look into it, I'll see if I can add that as part of this PR.

I addressed all the comments, except I'm still working on the coverage.

**Comment by maxlambrecht** [comment]
I configured a `build-and-test` workflow to run on push and pull_request events, using OS/Java combinations (Ubuntu, java 8, 11, 16 /  Macos,  java 8, 11, 14). Java 16 is not installed in Macos.

I also configured a `create-coverage` workflow to run when pushing , it sends the coverage repo to Coveralls, the same way we were doing. It run only on push because it needs a token that is retrieved from a Github Secret and `Secrets are not passed to workflows that are triggered by a pull request from a fork.`

I also updated the CI badge on the README.

File: .github/workflows/gradle.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      - name: Build with Gradle and generate the artifacts
+        run: ./gradlew build
+      - name: Cleanup Gradle Cache
+        # Remove some files from the Gradle cache, so they aren't cached by GitHub Actions.
+        # Restoring these files from a GitHub Actions cache might cause problems for future builds.
+        run: |
+          rm -f ~/.gradle/caches/modules-2/modules-2.lock
+          rm -f ~/.gradle/caches/modules-2/gc.properties
+
+  builad-and-test-on-macos:
```
Comment (rturner3) [inline .github/workflows/gradle.yml:43]:
```suggestion
  build-and-test-on-macos:
```

File: .github/workflows/gradle.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        java-version: [ 8, 11, 14] # macos-latest at his point doesn't have java 16 installed
```
Comment (rturner3) [inline .github/workflows/gradle.yml:48]: spacing nit
```suggestion
        java-version: [ 8, 11, 14 ] # macos-latest at his point doesn't have java 16 installed
```

**Review by rturner3 — APPROVED** [review]
Looks great, thanks for setting it up!

**Pull request merged**